panel/src/components/Forms/Toolbar/LinkDialog.vue

```diff
@@ -65,6 +61,27 @@ export default {
 				return `<${this.value.url}>`;
 			}
 		},
+		open(input, selection) {
+			this.parseDOM(selection);
+			this.$refs.dialog.open();
+		},
+		parseDOM(selection) {
+			if (selection?.length === 0) {
+				this.value.text = null;
+				this.value.url  = null;
+			}
```

Comment:
Just a quick thought while reading this on my phone: should probably return here to not run the regex still with no selection.

Reply:
You're right!